### DISCUSSION TITLE
error when installing.

### DISCUSSION THREAD

**Discussion by eerke2021** [astrandb/miele#597] [Q&A]
![image](https://github.com/user-attachments/assets/d4055e9f-4ea5-457b-9076-c5f4e4e81fc9)
what does that mean? maybe I've copied the key or secret wrong? Tried many passwords. Checked email in App.

**Comment by ajwatt** [comment]
EDIT: fixed links

I'm going to guess that your client_id and/or client_secret are somehow corrupted, as from a bad copy-paste. I've spent many hours wrestling with the same problem you have here, and I have finally resolved it today. Here is what I did:

1. Visit https://www.miele.com/developer/swagger-ui/ and click the "authorize" button in the lower right side. **Manually** type in your client id and client secret. Tick both of the scope boxes and click "authorize." Use your miele.com/Miele app credential to log in. From here, if you get the 401 error: invalid_client, then it's time to give up on that client_id and generate a new one. Skip to `#3` below.
2. If the authorization worked in `#1`, then your problem is in the application credential stored in HA. Follow the instructions [here](https://github.com/astrandb/miele/discussions/247)  to update it, being sure to **manually** type everything rather than copying and pasting. Re-add the Miele integration and everything should work.
3. To generate a new client_id, I just returned to https://www.miele.com/f/com/en/register_api.aspx and registered again but with a different email address (the same address as before didn't work). Most email services allow you to put a dot in the name part of your address, so for ajwatt@example.com I could do a.jwatt@example.com. The email address you use here is otherwise irrelevant and does not have to be the same as the one you use to log in to miele.com and the app.
4. In a few minutes you'll get the email to register for a new client_id and client_secret. Follow that process and **save the confirmation page** in your browser. You can also copy and paste it to a text file for convenience, but the HTML page is your backup.
5. Repeat `#1` above, and it should work this time. So go back to HA settings and **delete** the Miele application credential. Now re-add the Miele integration and follow the "wizard" to get everything working.

**Reply by DouglasUrner** [reply]
Thank you! I had the same problem and was able to solve it with your first two steps. Really appreciate you sharing your troubleshooting tips and how to re-enter the credentials.
